pennylane_lightning/lightning_qubit/_state_vector.py

```diff
@@ -221,14 +266,25 @@ def _apply_lightning(
             elif method is not None:  # apply specialized gate
                 param = operation.parameters
                 method(wires, invert_param, param)
+            elif self._operation_is_sparse(operation):
+                CSR_SparseHamiltonian = operation.sparse_matrix()
```

Comment:
We don't need this `CSR_SparseHamiltonian` if `op_adjoint_base` is controlled right? If so, I'd like to suggest moving it to the branch it is actually going to be used.

Reply:
OK. I will do it.

Reply:
Done!